vlm/src/main/scala/geotrellis/contrib/vlm/gdal/GDAL.scala

```diff
@@ -88,8 +88,11 @@ object GDAL {
       case _ => throw new Exception(s"Could not find equivalent GDALResampleMethod for: $method")
     }
 
-  def open(path: String): Dataset = {
-    val ds = gdal.Open(path, gdalconstConstants.GA_ReadOnly)
+  def open(path: String): Dataset =
+    open(VSIPath(path))
```

Comment:
This would fail on correct /vsi paths look like

```scala
  describe("getting VSI uri") {
    it("doesn't change vsi uri") {
      val path = "/vsizip//tmp/some/data/data.zip"
      VSIPath(path).vsiPath shouldBe path
    }
  }
```

```
[info] - doesn't change vsi uri *** FAILED *** (14 milliseconds)
[info]   "/vsizip//[vsizip//]tmp/some/data/data.z..." was not equal to "/vsizip//[]tmp/some/data/data.z..." (VSIPathSpec.scala:306)
```

Reply:
I think the thing to do is to have 
- `GDAL.openPath` pass the string to GDAL unmodified
- `GDAL.openUri` performing the praising, assuming that what you get is a URI
- `GDAL.open` perform best effort check if the path is URI or path and dispatch accordingly.

I'd tend to leave `VSIPath` out-of signature for `GDAL.open` because I would not expect the user to have to know about this type, its mostly a complex parsing function. That also makes it easier to refactor and change it when bugs are found.